fwprovider/nodes/resource_acme_certificate.go

```diff
@@ -0,0 +1,882 @@
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Unable to read certificate information",
+			fmt.Sprintf("Failed to retrieve the ordered certificate for node %s after multiple attempts: %s", nodeName, err.Error()),
+		)
+
+		return
+	}
+
+	// Update the state with certificate information
+	if err := r.updateModelFromCertificates(ctx, &plan, certificates); err != nil {
+		resp.Diagnostics.AddError(
+			"Unable to process certificate information",
+			fmt.Sprintf("An error occurred while processing certificate information: %s", err.Error()),
+		)
+
+		return
+	}
+
+	plan.ID = plan.NodeName
+
+	resp.Diagnostics.Append(resp.State.Set(ctx, &plan)...)
+}
+
+// Read reads the current certificate information for the node.
+func (r *acmeCertificateResource) Read(
+	ctx context.Context,
+	req resource.ReadRequest,
+	resp *resource.ReadResponse,
+) {
+	var state acmeCertificateModel
+
+	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	nodeName := state.NodeName.ValueString()
+	nodeClient := r.client.Node(nodeName)
+
+	// Read certificate information
+	certificates, err := nodeClient.ListCertificates(ctx)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Unable to read certificate information",
+			fmt.Sprintf("An error occurred while reading certificate information for node %s: %s", nodeName, err.Error()),
+		)
+
+		return
+	}
+
+	// Update the state with certificate information
+	if err := r.updateModelFromCertificates(ctx, &state, certificates); err != nil {
+		resp.Diagnostics.AddError(
+			"Unable to process certificate information",
+			fmt.Sprintf("An error occurred while processing certificate information: %s", err.Error()),
+		)
+
+		return
+	}
+
+	resp.Diagnostics.Append(resp.State.Set(ctx, &state)...)
+}
+
+// Update renews the certificate for the node.
+func (r *acmeCertificateResource) Update(
+	ctx context.Context,
+	req resource.UpdateRequest,
+	resp *resource.UpdateResponse,
+) {
+	var plan, state acmeCertificateModel
+
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
+	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	nodeName := plan.NodeName.ValueString()
+	nodeClient := r.client.Node(nodeName)
+
+	// Update node configuration if account or domains changed
+	if !plan.Account.Equal(state.Account) || !plan.Domains.Equal(state.Domains) {
+		if err := r.configureNodeACME(ctx, nodeClient, &plan); err != nil {
+			resp.Diagnostics.AddError(
+				"Unable to update node ACME settings",
+				fmt.Sprintf("An error occurred while updating ACME settings for node %s: %s", nodeName, err.Error()),
+			)
+
+			return
+		}
+	}
+
+	// Order a new certificate if force is true or other changes are made
+	force := proxmoxtypes.CustomBool(plan.Force.ValueBool())
+	orderReq := &nodes.CertificateOrderRequestBody{
+		Force: &force,
+	}
+
+	taskID, err := nodeClient.OrderCertificate(ctx, orderReq)
```

Comment:
looks like we'll send the order call regardless if there were changes or not, or value of the force flag.

should we add a condition here like `if plan.Force.ValueBool() || !plan.Account.Equal(state.Account) || !plan.Domains.Equal(state.Domains) {....`